File: 001-500/0168. Excel Sheet Column Title.java
// Easy
// Math, String
// Time:O(logn),Space:O(logn)
// https://leetcode.cn/problems/excel-sheet-column-title/

class Solution {
  public String convertToTitle(int columnNumber) {
      StringBuilder sb = new StringBuilder();
      while (columnNumber > 0) {
          int a0 = (columnNumber - 1) % 26 + 1;
          sb.append((char)(a0 - 1 + 'A'));
          columnNumber = (columnNumber - a0) / 26;
      }
      return sb.reverse().toString();
  }
}

File: 0001-0500/0168. Excel Sheet Column Title.py
# Time:O(log₂₆ n), Space:O(log₂₆ n)

class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        letters = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        res = []
        n = columnNumber
        while n > 0:
            n -= 1
            remainder = n % 26
            res.append(letters[remainder])
            n //= 26

        return "".join(reversed(res))
